packages/mambajs/src/solver.ts

```diff
@@ -43,30 +43,69 @@ export interface ISolveOptions {
 const solve = async (
   specs: Array<string>,
   channels: Array<string>,
+  installedCondaPackages: ISolvedPackages,
   logger?: ILogger
 ) => {
   let result: SolvedPackage[] | undefined = undefined;
   const solvedPackages: ISolvedPackages = {};
   try {
+    let installed: any = [];
+    if (Object.keys(installedCondaPackages).length) {
+      Object.keys(installedCondaPackages).map((filename: string) => {
+        const installedPkg = installedCondaPackages[filename];
+        let tmpPkg = {
+          ...installedPkg,
+          packageName: installedPkg.name,
+          build: installedPkg.build_string,
+          filename
+        };
+
+        installed.push(tmpPkg);
+      });
+    } else {
+      installed = undefined;
+    }
     const startSolveTime = performance.now();
-    result = await simpleSolve(specs, channels, PLATFORMS);
+    result = (await simpleSolve(
+      specs,
+      channels,
+      PLATFORMS,
+      installed
+    )) as SolvedPackage[];
     const endSolveTime = performance.now();
-
     if (logger) {
       logger.log(
         `Solving took ${(endSolveTime - startSolveTime) / 1000} seconds`
       );
     }
-
+    result.forEach((pkg: any) => {
+      const installedPkg = installedCondaPackages[pkg.filename];
+      if (installedPkg) {
+        logger?.log(`${installedPkg.name} has been already installed`);
+      }
+    });
     result.map((item: any) => {
-      const { filename, packageName, repoName, url, version, build } = item;
+      const {
+        filename,
+        packageName,
+        repoName,
+        url,
+        version,
+        build,
+        buildNumber,
+        depends,
+        subdir
+      } = item;
       solvedPackages[filename] = {
         name: packageName,
         repo_url: repoName,
         build_string: build,
         url: url,
         version: version,
-        repo_name: repoName
+        repo_name: repoName,
+        build_number: buildNumber,
+        depends,
+        subdir
       };
     });
```

Comment:
We should change our definition of `ISolvedPackage` so we don't need to do this. We should rename `build_number` -> `buildNumber`, `repo_name` -> `repoName` etc. 

Then no need for this loop?